python/aitemplate/frontend/nn/attention.py

```diff
@@ -130,7 +183,33 @@ def __init__(
         if seq_len % 2 == 1:
             self.use_flash = True
 
-        if use_mem_eff:
+        if use_vanilla or (USE_CUDA and detect_target()._arch in ["70"]):
+            if causal:
+                import torch
+
+                from aitemplate.compiler.base import _TorchConstantTensorData
+
+                mask = torch.triu(
+                    torch.ones(seq_len, seq_len, dtype=torch.bool), 1
+                ).cuda()
+                causal_mask_pt = torch.zeros(seq_len, seq_len).cuda().half()
+                causal_mask_pt.masked_fill_(mask, float("-inf"))
+                causal_mask_pt = causal_mask_pt.unsqueeze(0)
+
+                causal_mask = Tensor(
```

Comment:
can we move this part outside frontend? user can assign the mask tensor if needed.

Reply:
Makes sense, will create a new class and update the PR. 
The mask tensor constant is to implement "causal" so should be compile-time

Reply:
Yes, mask tensor can be assigned before or after model compiling. It'd be good to keep ait frontend clean and avoid pytorch related code.

Reply:
Created a separate class and now the mask should be passed.